src/clj/puppetlabs/services/file_serving/file_metadata/file_metadata_core.clj

```diff
@@ -0,0 +1,200 @@
+                :else (get-file-metadata file-path posix-attrs)) [:data :links] links)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Compojure app
+
+(defn try-to-parse
+  [body]
+  (try
+    (cheshire/parse-stream (io/reader body) true)
+    (catch Exception e
+      (log/debug e))))
+
+(def media-types
+  #{"application/json" "text/pson" "pson"})
+
+(defn content-type-valid?
+  [context]
+  (let [content-type (get-in context [:request :headers "content-type"])]
+    (or
+      (nil? content-type)
+      (media-types content-type))))
+
+(defn as-json-or-pson
+  "This is a stupid hack because of PSON.  We shouldn't have to do this, but
+  liberator does not know how to serialize a map as PSON (as it does with JSON),
+  so we have to tell it how."
+  [x context]
+  (let [context-with-media-type (if (string/blank? (get-in context
+                                                           [:representation
+                                                            :media-type]))
+                                  (assoc-in context
+                                            [:representation :media-type]
+                                            "text/pson")
+                                  context)]
+    (-> (cheshire/generate-string x)
+        (representation/as-response context-with-media-type)
+        (assoc :status 200)
+        (representation/ring-response))))
+
+(defresource file-metadata
+             [path service links source_permissions]
+             :allowed-methods [:get]
+
+             :allowed? (fn [context]
+                         (->> (fileserver-config/find-mount service path)
+                              (second)
+                              (fileserver-config/allowed? service (:request context))))
+
+             :available-media-types media-types
+
+             :handle-exception utils/exception-handler
+
+             :exists?
+             (fn [context]
+               (let [[mount path] (fileserver-config/find-mount service path)
+                     file (clojure.java.io/file (:path mount) path)]
+                 (.exists file)))
+
+             :handle-ok
+             (fn [context]
+               (-> (fileserver-config/find-mount service path)
+                   (get-metadata links source_permissions)
+                   (as-json-or-pson context)))
+
+             :new? false)
+
+
+(schema/defn routes
+  [fileserving-config-service]
+  (compojure/context "/:environment" [environment]
+                     (compojure/routes
+                       (GET "/file_metadata/*" request
+                            (let [path (get-in request [:params :*])
+                                  links (get-in request [:params :links] "manage")
+                                  permissions (get-in request [:params :use_permissions] "use")]
+                              (file-metadata path fileserving-config-service links permissions))))))
```

Comment:
With some of Compojure's magic, I think that you could destructure `request` as `{params :params`}, and then do e.g.
`links (get params :links "manage")`.